src/bynder-js-sdk.js

```diff
@@ -571,4 +573,327 @@ export default class Bynder {
+                });
+                form.append('file', chunkData);
+                const headers = Object.assign(form.getHeaders(), {
+                    'content-length': form.getLengthSync()
+                });
+                return axios.post(endpoint, form, { headers });
+            };
+
+            // sequentially upload chunks to AWS, then register them
+            let chunk = 0;
+            if (isBuffer) {
+                // buffer input
+                return new Promise((resolve, reject) => {
+                    (function nextChunk() {
+                        if (chunk < chunks) {
+                            const start = chunk * CHUNK_SIZE;
+                            const end = Math.min(start + CHUNK_SIZE, length);
+                            const chunkData = body.slice(start, end);
+                            uploadChunkToS3(chunkData, ++chunk)
+                                .then(() => {
+                                    return registerChunk(init, chunk);
+                                })
+                                .then(nextChunk)
+                                .catch(reject);
+                        } else {
+                            // pass init and chunk number to finalize
+                            resolve({ init, chunk });
+                        }
+                    }());
+                });
+            }
+
+            // readable stream input
+            return new Promise((resolve, reject) => {
+                (function nextChunk() {
+                    if (chunk < chunks) {
+                        const chunkData = body.read(CHUNK_SIZE);
+                        if (chunkData !== null) {
+                            uploadChunkToS3(chunkData, ++chunk)
+                                .then(() => {
+                                    return registerChunk(init, chunk);
+                                })
+                                .then(nextChunk)
+                                .catch(reject);
+                        } else {
+                            // our read stream is not done yet reading
+                            // let's wait for a while...
+                            setTimeout(nextChunk, 50);
+                        }
+                    } else {
+                        // pass init and chunk number to finalize
+                        resolve({ init, chunk });
+                    }
+                }());
+            });
+        })
+        .then((uploadResponse) => {
+            const { init, chunk } = uploadResponse;
+            return finalizeUpload(init, filename, chunk);
+        })
+        .then((finalizeResponse) => {
+            const { importId } = finalizeResponse;
+            return new Promise((resolve, reject) => {
+                const POLLING_INTERVAL = 2000;
+                const MAX_POLLING_ATTEMPTS = 60;
+                let attempt = 0;
+                (function checkStatus() {
```

Comment:
Also this function could be extracted from here, there's no need to have all these levels increasing the complexity of the code.

Reply:
Done in 111033ceca607f67883bd3abca10aae0a1e77b91